.github/workflows/build.sh

```diff
@@ -4,6 +4,9 @@ set -eux
 set -o pipefail
 
 export ASAN_UBSAN=${ASAN_UBSAN:-false}
+export WERROR=${WERROR:-false}
+
+additional_autogen_args="--enable-compat-howl --enable-compat-libdns_sd"
```

Comment:
I would leave those expandable by environment if possible. We have something here and something below at autogen.sh. Would it make more sense to have configure options only on single place?

Reply:
I think ideally all those flags should probably be passed via environment variables. The problem is that at this point due to various build system bugs it's kind of hard to run the script directly. For example, I can test `avahi` locally with `clang` and `-Werror` by running just
```sh
CC=clang WERROR=true ./.github/workflows/build.sh build
```
If I dropped the WERROR variable and relied on `CFLAGS` and so on I would have to copy-paste CFLAGS from the yml file and the command line would look like
```
CC=clang \
CFLAGS="-Werror -Wno-error=strict-prototypes -Wno-error=invalid-source-encoding -Wno-error=varargs -Wno-error=cast-align -Wno-error=logical-not-parentheses -Wno-error=deprecated-declarations" \
ADDITIONAL_AUTOGEN_ARGS="--disable-gobject --disable-compat-howl --disable-compat-libdns_sd" \
.github/workflows/build.sh build
```
The script would still have to look for `-Werror` in `CFLAGS` to be able to remove `common/acx_pthread.m4`.

In other words variables like `WERROR` and `ASAN_UBSAN` are used to get around a bunch of issues and once those issues are fixed I think it would probably make sense to replace them with just `CFLAGS=-Werror` and `CFLAGS=-fsanitize=address,undefined`.

Reply:
Trying to bring back the compat libraries I noticed that they trigger a lot of warnings and decided that it doesn't make sense to silence them everywhere so instead of using `CFLAGS` the script patches Makefiles on a case-by-case basis. I started using CFLAGS where I can though and at this point it's helpful when it comes to passing various optimization levels.

Reply:
I've just pushed two more commits where I fixed the "invalid-source-encoding" warnings and brought clang/ASan/UBsan.

Reply:
I've just push one more commit where I dropped the CodeQL autobuild script to fix the extraction of the Python code and the C# code.